src/progress/progress_set.rs

```diff
@@ -79,24 +91,15 @@ where
     }
 }
 
-impl From<ConfState> for Configuration {
-    fn from(conf_state: ConfState) -> Self {
+impl From<&ConfState> for Configuration {
```

Comment:
This looks really strange. How about add a `from_conf_state` method to `Configuration`?